tests/e2e/local/kind/README.md

```diff
@@ -0,0 +1,69 @@
+# Benefits:
+1. Set up a local kind(Kubernetes in Docker) environment once and run E2E tests on local machine, so you can test and debug locally.
+1. No need to worry about kubernetes cluster setup.
+
+# Prereqs:
+1. Set up Istio Dev environment using https://github.com/istio/istio/wiki/Dev-Guide.
+
+1. Install
+  * [docker](https://docs.docker.com/)
+  * [kind](https://github.com/kubernetes-sigs/kind)
+
+You can run the following script to check/install of all pre-requisites, or use it as a reference to install them manually.
+(This requires installation of [Homebrew](https://brew.sh) on MacOS or debian based Linux distributions)
+
+```bash
+. ./install_prereqs.sh
+```
+
+# Steps
+## 1. Set up kind Environment
+```bash
+. ./setup_host.sh
+```
+
+## 2. Build Istio images
+Build images on your host machine:
+```bash
+. ./setup_test.sh
+```
+
+## 2. Run tests!
+You can issue test commands on your host machine.
+E.g.
+```bash
+cd $ISTIO/istio
+make e2e_simple E2E_ARGS="--use_local_cluster" HUB=localhost:5000 TAG=latest
+```
+Note the special arguments like **E2E_ARGS**, **HUB**, and **TAG**. They are required to run these tests with the local cluster and a local registry inside the VM. And you can run multiple E2E tests sequentially against the same VM.
+The script has a number of options available [here](../../README.md#options-for-e2e-tests)
+
+# Cleanup
+To destroy the kind:
+```bash
+kind delete
+``` 
+
+### Debug with KubeSquash
+You can try debugging Istio with debugger tool [KubeSquash](https://github.com/solo-io/kubesquash). 
+For example, if you want to debug discovery container in pilot, follow steps as follows:
+1. Run that test in your host/vm.
+   ```bash
+   # In the VM/Host
+   make e2e_simple E2E_ARGS="--use_local_cluster --skip_cleanup" HUB=10.10.0.2:5000 TAG=latest
+   ```
+1. Run the kubesquash binary
+1. Select the namespace of istio mesh: istio-system
+1. Select Pilot Pod from the pods list
+1. Select discovery container after that
+1. You will get a prompt like `Going to attach dlv to pod istio-pilot-67db57c96d-c86ff. continue? `. Say Yes
+
+After this delve would be attached to discovery container for you to debug.
+
+For more information on debugging with delve, please check [Debug an Istio container with Delve](https://github.com/istio/istio/wiki/Dev-Guide#debug-an-istio-container-with-delve)
+
+# Troubleshooting
+Please refer [Troubleshooting](Troubleshooting.md) doc for information on this.
```

Comment:
writing trouble shooting and user guides is next up for me if I can work around https://github.com/kubernetes-sigs/kind/issues/43 soon. once we have those they'd be helpful here. in the meantime feel free to assign me to any questions..

Reply:
Thanks!